app/src/main/java/com/ggetters/app/ui/central/viewmodels/MatchEventViewModel.kt

```
@@ -0,0 +1,159 @@
+
+                            val inExisting = attendanceRepository.getById(event.matchId, playerInId)
+                            val inUpdated = (inExisting?.copy(status = 0)
+                                ?: com.ggetters.app.data.model.Attendance(
+                                    eventId = event.matchId,
+                                    playerId = playerInId,
+                                    status = 0,
+                                    recordedBy = "system"
+                                ))
+                            attendanceRepository.upsert(inUpdated)
+                            
+                            // Force immediate UI update by triggering roster refresh
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)